pygmt/tests/test_helpers.py

```diff
@@ -137,6 +138,17 @@ def test_gmttempfile_read():
         assert tmpfile.read(keep_tabs=True) == "in.dat: N = 2\t<1/3>\t<2/4>\n"
 
 
+@pytest.mark.parametrize(
+    "outfile", [123, "", ".", "..", "path/to/dir/", Path(), Path("..")]
```

Comment:
Check Windows-style paths also.

```suggestion
    "outfile", [123, "", ".", "..", "path/to/dir/", r"path\to\dir\\", Path(), Path(".."), ]
```